src/main/java/com/github/jikoo/regionerator/ChunkFlagger.java

```diff
@@ -6,120 +6,79 @@
 import com.google.common.cache.CacheLoader;
 import com.google.common.cache.LoadingCache;
 import com.google.common.cache.RemovalListener;
-import com.google.common.cache.RemovalNotification;
 import java.io.File;
-import java.io.IOException;
-import java.util.HashMap;
-import java.util.Iterator;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
 import java.util.Map;
 import java.util.concurrent.TimeUnit;
-import org.bukkit.Bukkit;
+import java.util.regex.Pattern;
+
 import org.bukkit.World;
 import org.bukkit.configuration.ConfigurationSection;
 import org.bukkit.configuration.file.YamlConfiguration;
-import org.bukkit.scheduler.BukkitTask;
 import org.jetbrains.annotations.NotNull;
 
 /**
  * Storing time stamps for chunks made easy.
- * 
+ *
  * @author Jikoo
  */
 public class ChunkFlagger {
 
 	private final Regionerator plugin;
-	private final LoadingCache<Pair<String, String>, Pair<YamlConfiguration, Boolean>> flagFileCache;
-	private final HashMap<Pair<String, String>, YamlConfiguration> saveQueue;
-	private BukkitTask saveTask = null;
+	private final LoadingCache<String, FlagData> flagFileCache;
+	private final Connection database;
 
 	ChunkFlagger(Regionerator plugin) {
 		this.plugin = plugin;
-		this.saveQueue = new HashMap<>();
+		try {
+			Class.forName("org.sqlite.JDBC").newInstance();
+			this.database = DriverManager.getConnection("jdbc:sqlite://" + plugin.getDataFolder().getAbsolutePath() + "/data.db");
+			try (Statement st = database.createStatement()) {
```

Comment:
Does SQLite support PreparedStatements? I know injection is not a problem here, but I've always favored a bit of overkill.

Reply:
Supports of course